pyrokinetics/readers.py

```diff
@@ -5,19 +5,31 @@
 """
 
 from pathlib import Path
-from typing import Union, Type, Any
+from typing import Optional, Union, Type, Any
 from .typing import PathLike
 from abc import ABC, abstractmethod
 from .factory import Factory
 
 
 class Reader(ABC):
     @abstractmethod
-    def read(self, filename: PathLike, *args, **kwargs) -> Any:
-        """Read and process a file"""
+    def read(
+        self,
+        filename: PathLike,
+        second_filename: Optional[PathLike] = None,
```

Comment:
Rather than adding a `second_filename` arg to the `Reader` class, I'd prefer to make it more generic and take an arbitrary number of filenames:

```python
def read(self, *filenames: PathLike, **kwargs) -> Any:
    ...
```
We ran into a similar problem when reading GENE output data, as that's also spread across multiple different files, and it required a bit of a hack to get things working. I'll try to make a new PR into this one sometime today which will hopefully make the base class generic enough to handle any other cases like this we might run into in the future.